pkg/admission/apiresourceschema/validation_test.go

```diff
@@ -23,12 +23,14 @@ import (
 
 func TestValidationOptionDrift(t *testing.T) {
 	expectedNonBool := map[string]reflect.Kind{
-		"DisallowDefaultsReason": reflect.String,
-		"CELEnvironmentSet":      reflect.Ptr,
-		"PreexistingExpressions": reflect.Struct,
+		"DisallowDefaultsReason":       reflect.String,
+		"CELEnvironmentSet":            reflect.Ptr,
+		"PreexistingExpressions":       reflect.Struct,
+		"VersionsWithUnchangedSchemas": reflect.Map,
 	}
 	expectedFalse := map[string]bool{
-		"RequireImmutableNames": true,
+		"RequireImmutableNames":     true,
+		"SuppressPerExpressionCost": true,
```

Comment:
remind me why this change is correct?

Reply:
I'm not 100% sure here, appreciate more input, but since it's a new field I thought that "expecting it to be false" was a reasonable assumption. This is the field in question:

```go
// versionsWithUnchangedSchemas tracks schemas of which versions are unchanged when updating a CRD.
// Does not apply to creation or deletion.
// Some checks use this to avoid rejecting previously accepted versions due to a control plane upgrade/downgrade.
versionsWithUnchangedSchemas sets.Set[string]
// suppressPerExpressionCost indicates whether CEL per-expression cost limit should be suppressed.
// It will be automatically set during Versions validation if the version is in versionsWithUnchangedSchemas.
suppressPerExpressionCost bool
```

I didn't see it as important that it's set.

Reply:
The test here is to force us understand which values new fields should have.

False is correct. Please add the reasoning to `defaultValidationOpts`:
```golang
	// in Kube, the validation on CRD update will set it to true for unchanged schemas
	// (ratcheting validation). Here, as everything is immutable, this case never
	// happens. Hence, we can statically set it to false.
	SuppressPerExpressionCost: false,
```